server/src/main/java/org/elasticsearch/common/util/CollectionUtils.java

```diff
@@ -145,6 +150,18 @@ public static boolean isEmpty(Object[] array) {
         return array == null || array.length == 0;
     }
 
+    /**
+     * Null-safe check if the specified collection is empty.
+     * <p>
+     * Null returns true.
+     *
+     * @param coll  the collection to check, may be null
+     * @return true if empty or null
+     */
+    public static boolean isEmpty(final Collection<?> coll) {
```

Comment:
I'd prefer not do have this and to do the null check at the call site to honest.